### PR TITLE
feat: My view report page to use the platform API for report expenses.

### PR DESCRIPTION
**Pull request by arjunaj5**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9489577</samp>

This pull request refactors the report view and the expenses service to use the new platform expense model and the new expense card component. It removes the unused imports and functions that depend on the old expense model and the old transaction model. It also improves the performance and ordering of the expenses service by using the `Cacheable` decorator and the order parameter.
https://app.clickup.com/t/86cty99qw

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9489577</samp>

> _We're refactoring the report view, me hearties, yo ho ho_
> _We're using the new platform model, me hearties, yo ho ho_
> _We're sorting and caching the expenses, me hearties, yo ho ho_
> _We're heaving on the `app-expense-card-v2`, on the count of three_

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9489577</samp>

*  Cache and sort report expenses using the Cacheable decorator and the order parameter ([link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7e1ed82416d113f0ad13f11d3238de7424dc72ac6b24ffe030dd7a66b2b139bfR84), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-43a6608834bfbe4963726f4de34dd6082b69581858020acfa23be1e61037d9f5R8-R9), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-43a6608834bfbe4963726f4de34dd6082b69581858020acfa23be1e61037d9f5L37-R45))
*  Use the new expense card component with the new platform expense model in the report view ([link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-a52aaafa54cc049d42c344ff3d466b8f46ad3b7aca50bda871b8f6258071f3d4L162-R172), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aR29), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL47-R50), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL87-R90), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL254-R254), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL282-R264), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL447-R434), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL478-R460), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL489-R473), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL533-R515))
*  Inject the new expenses service and use its functions and enums in the report view ([link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aR36-R37), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aR116), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL545-R528))
*  Remove unused imports and functions from the report service, the transaction service, and the report view ([link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L8), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-712edde8a2bf233766bbc21a32a8166358eca7a00851e1a25af976cff8b780b5L19), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-fe3424c28721c88278049c01f15468c2e33fa02bc82417ae132a2a7c5ae977bcL294-L312), [link](https://github.com/fylein/fyle-mobile-app/pull/2573/files?diff=unified&w=0#diff-7a0f8f3a8ecea2ebee0112759d8dccf6f71ac46657b799a08ce4b2ab836ac77aL166-L173))

## Clickup
Please add link here

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/transaction.service.ts [hunk 1]
```diff
@@ -16,7 +16,6 @@ import { Expense } from '../models/expense.model';
 import { Cacheable, CacheBuster } from 'ts-cacheable';
 import { UserEventService } from './user-event.service';
 import { UndoMerge } from '../models/undo-merge.model';
-import { AccountType } from '../enums/account-type.enum';
```
Comment (arjunaj5) [inline src/app/core/services/transaction.service.ts:19]: was not being used

File: src/app/core/services/report.service.ts [hunk 2]
```diff
@@ -5,7 +5,6 @@ import { catchError, concatMap, map, reduce, switchMap, tap } from 'rxjs/operato
 import { PAGINATION_SIZE } from 'src/app/constants';
 import { CacheBuster, Cacheable } from 'ts-cacheable';
 import { ApiV2Response } from '../models/api-v2.model';
-import { Expense } from '../models/expense.model';
```
Comment (arjunaj5) [inline src/app/core/services/report.service.ts:8]: was not being used

File: src/app/fyle/my-view-report/my-view-report.page.ts [hunk 3]
```diff
@@ -251,24 +247,10 @@ export class MyViewReportPage {
...
-            tx_report_id: 'eq.' + this.reportId,
-            order: 'tx_txn_dt.desc,tx_id.desc',
-          })
-          .pipe(finalize(() => (this.isExpensesLoading = false)))
-      ),
-      map((etxns) =>
-        etxns.map((etxn) => {
-          etxn.vendor = this.getVendorName(etxn);
-          etxn.violation = this.getShowViolation(etxn);
-          return etxn;
```
Comment (arjunaj5) [inline src/app/fyle/my-view-report/my-view-report.page.ts:270]: removed as the `etxn.vendor` and `etxn.violation` was not being used anywhere else.

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 4]
```diff
@@ -5,6 +5,8 @@ import { PlatformApiResponse } from 'src/app/core/models/platform/platform-api-r
 import { Expense } from 'src/app/core/models/platform/v1/expense.model';
```
Comment (arjunaj5) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:5]: Will be fixing up tests in another PR: https://app.clickup.com/t/86cty99r6

File: src/app/core/services/platform/v1/spender/expenses.service.ts [hunk 5]
```diff
@@ -5,6 +5,8 @@ import { PlatformApiResponse } from 'src/app/core/models/platform/platform-api-r
...
 import { ExpensesQueryParams } from 'src/app/core/models/platform/v1/expenses-query-params.model';
 import { PAGINATION_SIZE } from 'src/app/constants';
+import { Cacheable } from 'ts-cacheable';
+import { expensesCacheBuster$ } from '../../../transaction.service';
```
Comment (arjunaj5) [inline src/app/core/services/platform/v1/spender/expenses.service.ts:9]: ref [thread](https://fylein.slack.com/archives/C060N8C1X0R/p1700114834471279), was thinking if we should move this to this file only.

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/my-view-report/my-view-report.page.ts [hunk 6]
```diff
@@ -542,8 +524,8 @@ export class MyViewReportPage {
     this.trackingService.clickViewReportInfo({ view: ExpenseView.individual });
   }
 
-  canEditTxn(txState: string): boolean {
-    return this.canEdit$ && ['DRAFT', 'DRAFT_INQUIRY', 'COMPLETE', 'APPROVER_PENDING'].indexOf(txState) > -1;
+  canEditExpense(expenseState: ExpenseState): boolean {
+    return this.canEdit$ && ['DRAFT', 'DRAFT_INQUIRY', 'COMPLETE', 'APPROVER_PENDING'].indexOf(expenseState) > -1;
```
[the remark this answers is not in the document]
Comment (arjunaj5) [reply src/app/fyle/my-view-report/my-view-report.page.ts:528]: Yes, [ref thread](https://fylein.slack.com/archives/C060N8C1X0R/p1700112368116209), will put this in a separate PR 🙇

**Pull request merged**